hass-addon-sunsynk-dev/run.py

```diff
@@ -203,18 +239,36 @@ def setup_sensors() -> None:
         else:
             msg[fstr].append(name)  # type: ignore
 
-        SENSORS.append(getfilter(fstr, sensor=sen))
+        filt = getfilter(fstr, sensor=sen)
+        SENSORS.append(filt)
+        filters[sen.id] = filt
 
         if isinstance(sen, NumberRWSensor):
-            startup_sens.update(d.id for d in sen.dependencies())
+            for dep in sen.dependencies():
+                sens_dependencies[dep.id].append(sen)
```

Comment:
Note that we're constructing a reverse map of dependencies here.
So if sensor A depends on sensor S and sensor B depends on sensor S, then we will have S with two dependants - A and B.

Reply:
:+1: maybe call it sens_dependants?

Reply:
100%, I wanted to call it exactly that after my first commit but I had commented already and didn't want to push a new commit which would mess up the comments.